Api/Controllers/CompensationController.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Application.Dtos;
+using Application.Services;
+using Core;
+using TourmalineCore.AspNetCore.JwtAuthentication.Core.Filters;
+using Application;
+
+namespace Api.Controllers;
+
+[Authorize]
+[Route("api/compensations")]
+public class CompensationController : Controller
+{
+    private readonly CompensationsService _compensationsService;
+    private readonly InnerCircleHttpClient _client;
+
+    public CompensationController(CompensationsService compensationsService, InnerCircleHttpClient client)
+    {
+        _compensationsService = compensationsService;
+        _client = client;
+    }
+
+    [RequiresPermission(UserClaimsProvider.CanRequestCompensations)]
+    [HttpPost("create")]
+    public async Task CreateAsync([FromBody] CompensationCreateDto dto)
+    {
+        var employee = await _client.GetEmployeeAsync(User.GetCorporateEmail());
+        await _compensationsService.CreateAsync(dto, employee);
+    }
+
+    [RequiresPermission(UserClaimsProvider.CanRequestCompensations)]
+    [HttpGet("all")]
+    public async Task<PersonalCompensationListDto> GetAllAsync()
+    {
+        var employee = await _client.GetEmployeeAsync(User.GetCorporateEmail());
+        return await _compensationsService.GetAllAsync(employee);
+    }
+
+    [RequiresPermission(UserClaimsProvider.CanRequestCompensations)]
+    [HttpGet("types")]
+    public async Task<List<CompensationType>> GetTypesAsync()
+    {
+        return await _compensationsService.GetTypesAsync();
```

Comment:
The method `GetTypesAsync` is not a task. It just returns a list. Remove "await"